File: app/src/main/java/com/bzh/dytt/data/db/AppDatabase.java
package com.bzh.dytt.data.db;


import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;
import android.support.annotation.VisibleForTesting;

import com.bzh.dytt.data.CategoryMap;
import com.bzh.dytt.data.CategoryPage;
import com.bzh.dytt.data.VideoDetail;

@Database(entities = {VideoDetail.class, CategoryMap.class, CategoryPage.class}, version = 2)
@TypeConverters({DataTypeConverter.class})
public abstract class AppDatabase extends RoomDatabase {

    @VisibleForTesting
    public static final String DATABASE_NAME = "dytt-data.db";

    public abstract VideoDetailDAO videoDetailDAO();

    public abstract CategoryMapDAO categoryMapDAO();

    public abstract CategoryPageDAO categoryPageDAO();


}

File: app/src/main/java/com/bzh/dytt/data/db/AppDatabase.kt
package com.bzh.dytt.data.db

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.TypeConverters
import android.support.annotation.VisibleForTesting
import com.bzh.dytt.data.entity.CategoryMap
import com.bzh.dytt.data.entity.CategoryPage
import com.bzh.dytt.data.entity.VideoDetail

@Database(
        entities = [VideoDetail::class, CategoryMap::class, CategoryPage::class],
        version = 2
)
@TypeConverters(DataTypeConverter::class)
abstract class AppDatabase : RoomDatabase() {

    abstract fun videoDetailDAO(): VideoDetailDAO

    abstract fun categoryMapDAO(): CategoryMapDAO

    abstract fun categoryPageDAO(): CategoryPageDAO

    companion object {
        @VisibleForTesting
        val DATABASE_NAME = "dytt-data.db"
    }
}
